### PR TITLE
Correcting Export GEO 

### PR DESCRIPTION
**Pull request by Snehal1894**
Added the `if condition` for `Enzyme name` and `choice name` in the code.
This is where it was giving error.

### CODE REVIEW

File: Docker/development_docker-compose.yml [hunk 1]
```diff
@@ -3,6 +3,9 @@ version: '2'
 services:
   db:
     image: postgres:9.5
+    environment:
```
Comment (sergpolly) [inline Docker/development_docker-compose.yml:6]: @Snehal1894 what is this for could comment briefly ?

File: cLIMS/organization/export.py [hunk 2]
```diff
@@ -1225,7 +1225,12 @@ def removeDup(dcicExcelSheet):
                 
 @login_required 
 def exportDCIC(request,prj_pk):
-    expPks = request.POST.getlist('dcic')
+    input_keys = [key for key in request.POST if key.startswith("dcic")]
```
Comment (sergpolly) [inline cLIMS/organization/export.py:1228]: @Snehal1894 that is the fix for the DCIC issue that @betulakgol mentioned , right ?

what's the nature of the bug ?

File: cLIMS/organization/migrations/0001_initial.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
+# -*- coding: utf-8 -*-
```
Comment (sergpolly) [inline cLIMS/organization/migrations/0001_initial.py:1]: I've read in that django book - that updating the whole thing by the mean of migrations is a good thing ...

But I think @nandankita wasn't using them before - I wonder if that would break anything ? @Snehal1894 did you test it locally ?

File: cLIMS/organization/migrations/0002_auto_20200529_1432.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            model_name='award',
+            name='award_exp',
+            field=models.ManyToManyField(related_name='awardExp', to='organization.Experiment'),
+        ),
+        migrations.AddField(
+            model_name='award',
+            name='award_project',
+            field=models.ManyToManyField(related_name='awardPro', to='organization.Project'),
+        ),
+    ]
```
Comment (sergpolly) [inline cLIMS/organization/migrations/0002_auto_20200529_1432.py:110]: question about the file below - DCIC-Metadata_entry_form_FEB2018.xlsx - is it a new form ? do we keep forms as a part of the github repo ? - If that's the case - than ok

File: cLIMS/organization/static/siteWide/js/show.js [hunk 5]
```diff
@@ -99,10 +99,10 @@ $(document).ready(function(){
        });
         $("#unselect").click(function(){
             alert("If nothing selected all experiments will be considered.")
-            $('input:checkbox[name=dcic]').attr('checked',false);
+            $('input:checkbox').attr('checked',false);
```
Comment (sergpolly) [inline cLIMS/organization/static/siteWide/js/show.js:102]: is that just removing some default and nothing else  ?

File: cLIMS/organization/templates/dcicView.html [hunk 6]
```diff
@@ -9,40 +9,53 @@
...
 <span class="tableHead">
-	Experiment Details
+  Experiment Details
 </span>
 {% if messages %}
         {% for message in messages %}
             <p class="alert alert-danger">{{message}}</p>
         {% endfor %}
 {% endif %}
+<br/>**If nothing is selected by default it will take all experiments<br/>
```
Comment (sergpolly) [inline cLIMS/organization/templates/dcicView.html:21]: wow ! that a lot of changes @Snehal1894 - that is that DCIC related bug reported by @betulakgol , right ?

could you at least briefly describe what's going on here ?

File: cLIMS/organization/migrations/0001_initial.py (hunk 3 above)
Comment (sergpolly) [reply cLIMS/organization/migrations/0001_initial.py:1]: i think it has to do with the state of the database, right ? i.e. `models` describe how the underlying database is organized

File: cLIMS/organization/views.py [hunk 7]
```diff
@@ -1616,10 +1616,15 @@ class DcicView(View):
     def get(self,request,prj_pk):
         projectId = prj_pk
         context = {}
+        sets={}
         project = Project.objects.get(pk=projectId)
-        experiments=Experiment.objects.filter(project=projectId).order_by('-pk')
+        #experiments=Experiment.objects.filter(project=projectId).order_by('-pk')
+        expSet = ExperimentSet.objects.filter(project=projectId).order_by('-pk')
```
Comment (sergpolly) [inline cLIMS/organization/views.py:1622]: @Snehal1894 what's up here ? something with `experiementset`-s ?

which problem is it related to - DCIC or GEO ?
is this related to the new feature that you were describing to Job ? - like DCIC exporting experiment sets or something like that ?

File: cLIMS/wetLab/forms.py [hunk 8]
```diff
@@ -84,7 +84,7 @@ class IndividualForm(ModelForm):
         #add_related_field_wrapper(self, 'documents')
     class Meta:
         model = Individual
-        exclude = ('individual_fields','userOwner','dcic_alias','update_dcic','temp_var')
+        exclude = ('individual_fields','userOwner','dcic_alias','update_dcic')
```
Comment (sergpolly) [inline cLIMS/wetLab/forms.py:87]: @Snehal1894 what's up with `temp_var` what was it ?

File: Docker/development_docker-compose.yml (hunk 1 above)
Comment (Snehal1894) [reply Docker/development_docker-compose.yml:6]: There are updates in Postgres 9.x images available on Docker Hub, for security purposes. Now we either need to set password for specific user to get the connect or we can explicity mention that we allow all connection. We have done the later.
Git hub link for this change is : https://github.com/docker-library/postgres/pull/658

File: cLIMS/organization/export.py (hunk 2 above)
Comment (Snehal1894) [reply cLIMS/organization/export.py:1228]: Yes, Initially in ExportDCIC, html page showed all experiments and if the selected exps were not in any Experiment set, it would not generate DCIC excel sheet
So we are now displaying Experiment Sets and the corresponding experiments on html page.
So the code is modified to handle experiments according to the Experiment Sets.

File: cLIMS/organization/migrations/0001_initial.py (hunk 3 above)
Comment (Snehal1894) [reply cLIMS/organization/migrations/0001_initial.py:1]: Yes. And our code on git hub doesn't have these files. They are automatically updated according to our changes (or we generate new migration files in case there are changes wherever we are running the code). It is better to do it on server I think. So we don't need to add these type of files

File: cLIMS/organization/migrations/0002_auto_20200529_1432.py [hunk 9]
```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
```
Comment (Snehal1894) [inline cLIMS/organization/migrations/0002_auto_20200529_1432.py:1]: Also this should be deleted

File: cLIMS/organization/static/siteWide/js/show.js (hunk 5 above)
Comment (Snehal1894) [reply cLIMS/organization/static/siteWide/js/show.js:102]: Intially the DCIC html page has just one table with all exps and all checkbox fields had name "dcic".
But now there are multiple tables on this page, consisting of exps for every expset.
In order to have (un)select all for each table of Exp Set, so the names are modified to include Exp Set id ( dcic(expset.id)" like dcic77)

Also there is a button which should unselect al checkboxes, so I modified the code based on the "input type" to access all boxes on the page, and unselect them.

File: cLIMS/organization/templates/dcicView.html (hunk 6 above)
Comment (Snehal1894) [reply cLIMS/organization/templates/dcicView.html:21]: The HTML page is modified to display tables containing exps for each experiment set (expset).
Expset is a dictionary {Expset object: [all exp objects]}.
So for every Experiment Set (key in dictionary) a table is generated , with experiments in list (value of dictionary)

File: cLIMS/organization/views.py (hunk 7 above)
Comment (Snehal1894) [reply cLIMS/organization/views.py:1622]: Yes. This is related to DCIC update. I needed the ExperimentSet Objects and also exps in it to send to html page.
so here dictionary is created  sets.
It has {Expset object: [all exp objects]}

File: cLIMS/wetLab/forms.py (hunk 8 above)
Comment (Snehal1894) [reply cLIMS/wetLab/forms.py:87]: While trying to work with fixing the multiple tabs issue, I added this temporary hidden field. But we didn't use this approach. So I forgot to delete this field from the model. It is not needed.

File: Docker/development_docker-compose.yml (hunk 1 above)
Comment (sergpolly) [reply Docker/development_docker-compose.yml:6]: #73

File: cLIMS/organization/migrations/0001_initial.py (hunk 3 above)
Comment (sergpolly) [reply cLIMS/organization/migrations/0001_initial.py:1]: ok then I think we should remove them ...

File: cLIMS/organization/migrations/0002_auto_20200529_1432.py (hunk 9 above)
Comment (sergpolly) [reply cLIMS/organization/migrations/0002_auto_20200529_1432.py:1]: should be removed

File: cLIMS/organization/migrations/0002_auto_20200529_1432.py (hunk 4 above)
Comment (sergpolly) [reply cLIMS/organization/migrations/0002_auto_20200529_1432.py:110]: @Snehal1894 how about this one ? are we keeping those as part of the github repository ?

File: cLIMS/organization/views.py (hunk 7 above)
Comment (sergpolly) [reply cLIMS/organization/views.py:1622]: @Snehal1894 could you remove the commented code then ?

old code will remain in the history anyways

File: cLIMS/organization/migrations/0002_auto_20200529_1432.py (hunk 4 above)
Comment (Snehal1894) [reply cLIMS/organization/migrations/0002_auto_20200529_1432.py:110]: No, we need to update this excel sheet. The old sheet has existing entries ( that was the main issue, because of which the extra entries were seen). We need to update this on the server.

File: cLIMS/wetLab/models.py [hunk 10]
```diff
@@ -126,7 +126,7 @@ class Individual(References, UserOwner):
     dcic_alias = models.CharField(max_length=500, null=False, default="", unique=True, db_index=True, help_text="Provide an alias name for the object for DCIC submission.")
     update_dcic = models.BooleanField(default=False, help_text="This object needs to be updated at DCIC.")
     contributing_labs = models.ManyToManyField('organization.ContributingLabs', blank=True, help_text="Contributing labs for this individual.")
-    temp_var = models.IntegerField(null=True,blank=True)
+    #temp_var = models.IntegerField(null=True,blank=True)
```
Comment (sergpolly) [inline cLIMS/wetLab/models.py:129]: @Snehal1894  could you remove commented code please ?

File: cLIMS/organization/migrations/0002_auto_20200529_1432.py (hunk 4 above)
Comment (sergpolly) [reply cLIMS/organization/migrations/0002_auto_20200529_1432.py:110]: ok, so do we need to remove it from the repository ?

is this automatically generated by cLIMS or is it some form that one get from DCIC ?

File: cLIMS/organization/views.py (hunk 7 above)
Comment (Snehal1894) [reply cLIMS/organization/views.py:1622]: Yes yes, we can remove the commented lines.

File: cLIMS/organization/migrations/0002_auto_20200529_1432.py (hunk 4 above)
Comment (Snehal1894) [reply cLIMS/organization/migrations/0002_auto_20200529_1432.py:110]: Generated by cLIMS. Nothing directly related to DCIC. Automatically considers our updates in code

**Comment by sergpolly** [comment]
Awesome stuff @Snehal1894 !
